src/controller/CHIPDevice.cpp

```diff
@@ -171,14 +171,15 @@ CHIP_ERROR Device::Serialize(SerializedDevice & output)
 
     CHIP_ZERO_AT(serializable);
 
-    serializable.mOpsCreds   = mPairing;
-    serializable.mDeviceId   = Encoding::LittleEndian::HostSwap64(mDeviceId);
-    serializable.mDevicePort = Encoding::LittleEndian::HostSwap16(mDeviceUdpAddress.GetPort());
-    serializable.mAdminId    = Encoding::LittleEndian::HostSwap16(mAdminId);
-    SuccessOrExit(error = Inet::GetInterfaceName(mDeviceUdpAddress.GetInterface(), Uint8::to_char(serializable.mInterfaceName),
+    serializable.mOpsCreds        = mPairing;
+    serializable.mDeviceId        = Encoding::LittleEndian::HostSwap64(mDeviceId);
+    serializable.mDevicePort      = Encoding::LittleEndian::HostSwap16(mDeviceAddress.GetPort());
+    serializable.mAdminId         = Encoding::LittleEndian::HostSwap16(mAdminId);
+    serializable.mDeviceTransport = mDeviceAddress.GetTransportType();
```

Comment:
`chip::Transport::Type` is an enum, and probably will fit in a uint8_t. If we are assuming that, maybe we should check with a static assert that it does in fact fit into a uint8_t. Or enforce the endianness. Mostly for future proofing the code.